### PR TITLE
Enable Global Intermediate Buffers

### PR DESCRIPTION
**Pull request by rdspring1**
- [X] When the user designates a intermediate TensorView with the Global memory type, the fuser will pass the intermediate value as an argument to the kernel via global memory.
- [X] Enable support for the default schedule.

### CODE REVIEW

**Comment by rdspring1** [comment]
I made changes so that local memory is the default type and the memory type matches between the TensorView and its corresponding allocate node.

**Comment by csarofeen** [comment]
Could you also add/check there are couple things:
In addOutput/Input in fusion add a check to make sure they're in global memory
In the function to change memory type of a TV make sure we don't change a TV registered as input/output to anything besides global

File: torch/csrc/jit/codegen/cuda/fusion.cpp [hunk 1]
```diff
@@ -261,34 +261,43 @@ void Fusion::addInput(Val* const input) {
...
     auto tv = input->as<TensorView>();
-    if (tv->hasReduction())
+    if (tv->hasReduction()) {
       TORCH_WARN_ONCE(
           "Registered input ",
           input,
           " has a reduction axis, but this does nothing in the fusion.");
+    }
+    tv->setMemoryType(MemoryType::Global);
+    TORCH_INTERNAL_ASSERT(
```
Comment (csarofeen) [inline torch/csrc/jit/codegen/cuda/fusion.cpp]: Can this assert be removed as you just set it?

File: torch/csrc/jit/codegen/cuda/fusion.cpp [hunk 2]
```diff
@@ -261,34 +261,43 @@ void Fusion::addInput(Val* const input) {
...
       // Go to the root as we can merge bcast and
       // non-bcast dims, making a non-bcast dim.
-      TORCH_CHECK( // Should we warn instead?
+      TORCH_INTERNAL_ASSERT( // Should we warn instead?
           false,
           output,
           " cannot be registered as an output as it has a broadcast axis.");
+    }
+    tv->setMemoryType(MemoryType::Global);
+    TORCH_INTERNAL_ASSERT(
```
Comment (csarofeen) [inline torch/csrc/jit/codegen/cuda/fusion.cpp]: Do we need this assert?

**Review by csarofeen — APPROVED** [review]

File: torch/csrc/jit/codegen/cuda/fusion.cpp (hunk 1 above)
Comment (rdspring1) [reply torch/csrc/jit/codegen/cuda/fusion.cpp]: done.

File: torch/csrc/jit/codegen/cuda/fusion.cpp (hunk 2 above)
Comment (rdspring1) [reply torch/csrc/jit/codegen/cuda/fusion.cpp]: done

**Pull request merged**